Tests/IntegrationTests/Roslyn/Resources/PluginInspector.java

```diff
@@ -0,0 +1,296 @@
+import org.sonar.api.SonarPlugin;
+import org.sonar.api.config.PropertyDefinition;
+import org.sonar.api.server.rule.RulesDefinition;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
+import java.util.List;
+import java.util.jar.Attributes;
+import java.util.jar.Manifest;
+
+/**
+ * Simple console app used to test generated SonarQube plugin jar files.
+ * The app attempts to exercise the specified jar file by loading it
+ * and asking the plugin class for the defined extensions.
+ * An xml file is created containing information about the plugin e.g.
+ * the contents of the manifest and details of the extensions.
+ * Inputs:
+ *  1) Path to the jar file (required)
+ *  2) Full path of the xml file containing the output (optional)
+ *      If the second argument is not supplied then the output file name
+ *      is the jar file suffixed with ".dump.xml".
+ */
+public class PluginInspector  {
+
+    private static final int SuccessCode = 0;
+    private static final int ErrorCode = 1;
+
+    public static void main(String[] args) throws Exception {
+
+        Log("Starting Plugin inspector v0.1...");
+        String jarPath = null;
+
+        if (args.length != 1 && args.length != 2){
+            Log("Expecting at least one argument: the full path to the jar file.\r\nOptionally, the name of the file to be created may also be supplied.");
+            jarPath = "c:\\data\\code\\java\\jars\\Wintellect.Analyzers-plugin-1.0.5.jar";
+        }
+        else{
+            jarPath = args[0];
+        }
+
+        File file = new File(jarPath);
+        if (!file.exists()){
+            Log("The specified file does not exist: " + jarPath);
+            System.exit(ErrorCode);
+        }
+
+        Document doc = ProcessJar(jarPath);
+
+        int exitCode = SuccessCode;
+        if (doc == null) {
+            exitCode = ErrorCode;
+        }
+        else {
+            String xmlFilePath = args.length == 2 ? args[1] : jarPath + ".dump.xml";
+            SaveToFile(doc, xmlFilePath);
+        }
+
+        System.exit(exitCode);
+    }
+
+    private static Document ProcessJar(String jarPath) throws MalformedURLException, ClassNotFoundException {
+        Manifest manifest = TryGetManifest(jarPath);
```

Comment:
Try to load the manifest resource and add the contents to the XML file that is being built.
The manifest should contain the name of the plugin class we need to load i.e. the plugin entry point.